extension/icu/third_party/icu/common/ustrtrns.cpp

```diff
@@ -608,7 +608,7 @@ u_strFromUTF8Lenient(UChar *dest,
                 if((t1 = pSrc[1]) != 0 && (t2 = pSrc[2]) != 0 && (t3 = pSrc[3]) != 0) {
                     pSrc += 4;
                     /* 0x3c82080 = (0xf0 << 18) + (0x80 << 12) + (0x80 << 6) + 0x80 */
-                    ch = (ch << 18) + (t1 << 12) + (t2 << 6) + t3 - 0x3c82080;
+                    ch = (ch << 18) + ((UChar32)t1 << 12) + (t2 << 6) + t3 - 0x3c82080;
```

Comment:
`uint8_t t1, t2, t3; /* trail bytes */`

`t1 << 12` is causing undefined behavior because there aren't 12 bits in the value, so it has to be cast

Reply:
Apparantly this doesn't matter
`No arithmetic operation takes uint8_t so both of your operands will always be promoted.`

So this explicit cast is not necessary, it happens implicitly